README.md

```diff
@@ -41,9 +43,9 @@ await client.sendData(IlpPacket.serializeIlpPrepare({
 Two functions must be defined in order for the plugin to handle money.
 
 * `sendMoney (amount: string) -> Promise<null>`: sends `amount` of units to the peer. This should be done via a BTP `TRANSFER` call.
-* `handleMoney (from: null, { requestId: number, data: { amount: string, protocolData: Array<ProtocolData> } }) -> Array<ProtocolData>`: This function is called on an incoming BTP `TRANSFER`.
+* `_handleMoney (from: string, btpPacket: BtpPacket) -> Promise<Array<BtpSubProtocol>>`: This function is called on an incoming BTP `TRANSFER`.
```

Comment:
Why change `from` to a string? There is only one from, right?
Why pass a BtpPacket instead of separate requestId, amount, and protocolData? Unpacking that can be handled by the base class, right? Also, the BtpPacket already needs to be unpacked to see if it's a `Transfer` call, and so by the time `_handleMoney` is called, the unpacking work has already been done, would be a waste to do it a second time.

Reply:
> Why change from to a string?

It needs to have a type in TypeScript. The Typescript version uses an empty String instead of null so that it's still falsy.

> Why pass a BtpPacket instead of separate requestId, amount, and protocolData? Unpacking that can be handled by the base class, right?

I have intentionally made the method signatures backward compatible so that the JS generated by this should still work.

> Also, the BtpPacket already needs to be unpacked to see if it's a Transfer call, and so by the time _handleMoney is called, the unpacking work has already been done, would be a waste to do it a second time.

This is not encoding/decoding. BtpPacket is just a JS object with named and typed members. (i.e. A TypeScript interface) that match the original member names from the JS